### PR TITLE
adding whoami command

### PR DESCRIPTION
**Pull request by rishabh-vasudevan**
i have been able to print cluster adress, version and default app(if any). I have also added the --token argument but I am not able check whether the argument --token was there or not in the command. Could you please help me with the syntax of adding a check of whether the argument --token was present or not.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -126,6 +126,23 @@ fn main() -> Result<()> {
...
+        } else {
+            println!("cluster adress : {}", context.drogue_cloud_url);
+            println!(
+                "Default App : {}",
+                match context.default_app {
+                    None => "No default app",
+                    Some(ref x) => x,
+                }
+            );
+            util::print_version(&Ok(config));
```
Comment (jbtrystram) [inline src/main.rs:141]: Please move this logic to the `util` module. main is already quite long and messy so it's better if we keep only the arg-related suff here IMO.

File: src/main.rs [hunk 2]
```diff
@@ -126,6 +126,23 @@ fn main() -> Result<()> {
         exit(0);
     }
 
+    if matches.is_present(Other_commands::whoami) {
+        if let Some(_value) = matches.subcommand_matches("token") {
+            openid::print_token(&context);
```
Comment (jbtrystram) [inline src/main.rs:131]: I think what you look for is : 
```suggestion
        if matches.is_present("token") {
            openid::print_token(&context);
```

**Review by jbtrystram — CHANGES_REQUESTED** [review]

**Comment by rishabh-vasudevan** [comment]
I have made the changes. Please let me know if I should have done something differently.

**Review by jbtrystram — CHANGES_REQUESTED** [review]

File: src/arguments.rs [hunk 3]
```diff
@@ -89,7 +90,7 @@ pub fn parse_arguments() -> ArgMatches<'static> {
         .conflicts_with(Parameters::spec.as_ref())
         .help("file that contains the spec to update the resource with.");
 
-    let token_arg = Arg::with_name(Other_commands::token.as_ref())
+    let _token_arg = Arg::with_name(Other_commands::token.as_ref())
```
[the remark this answers is not in the document]
Comment (rishabh-vasudevan) [reply src/arguments.rs:93]: I did that by mistake. I was trying to remove the warning that was coming and added the _ absentmindedly.

File: src/arguments.rs [hunk 4]
```diff
@@ -40,6 +40,7 @@ pub enum Other_commands {
     login,
     token,
```
Comment (jbtrystram) [inline src/arguments.rs:41]: Can you also update the `token` command and mark is at deprecated ? Pointing to the new command would also be helpful for users :)

File: src/arguments.rs [hunk 5]
```diff
@@ -217,6 +218,16 @@ pub fn parse_arguments() -> ArgMatches<'static> {
...
                 .about("Print a valid bearer token for the drogue cloud instance."),
         )
+        .subcommand(
+            SubCommand::with_name(Other_commands::whoami.as_ref())
+                .about("Print cluster adress, version and default app(if any)")
+                .arg(
+                    token_arg
+                        .clone()
+                        .takes_value(false)
+                        .help("print a valid bearer toke for the drogue cloud instance."),
```
Comment (jbtrystram) [inline src/arguments.rs:228]: token

**Review by jbtrystram — APPROVED** [review]

**Pull request merged**